### PR TITLE
Pass PR info into event creation

### PR DESCRIPTION
**Pull request by DekusDenial**
blocked by: https://github.com/screwdriver-cd/data-schema/pull/323
related: https://github.com/screwdriver-cd/screwdriver/issues/1528

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -21,9 +21,24 @@ const TEMPORAL_JWT_TIMEOUT = 12 * 60; // 12 hours in minutes
  * @return {Array}            Array of job ids that match the job names
  */
 function findIdsOfMatchedJobs(jobs, jobNames) {
-    const matchedJobs = jobs.filter(j => jobNames.includes(j.name));
+    return jobs.reduce((match, j) => {
+        let name = j.name;
 
-    return matchedJobs.map(j => j.id);
+        // Get the actual job name if the matching job is a PR job
+        if (j.isPR) {
```
Comment (d2lam) [inline lib/build.js]: I think this needs to be `j.isPR()`
it's a function

File: lib/eventFactory.js [hunk 2]
```diff
@@ -175,7 +175,15 @@ function startBuild(config) {
...
 function createBuilds(config) {
-    const { eventConfig, eventId, pipeline, pipelineConfig, changedFiles, webhooks } = config;
+    const {
+        eventConfig,
+        eventId,
+        pipeline,
+        pipelineConfig,
+        changedFiles,
+        webhooks,
+        prInfo
```
Comment (d2lam) [inline lib/eventFactory.js]: Need to change jsdoc to include `prInfo`

File: lib/eventFactory.js [hunk 3]
```diff
@@ -119,13 +119,13 @@ function startBuild(config) {
     const BuildFactory = require('./buildFactory');
     const buildFactory = BuildFactory.getInstance();
     /* eslint-enable global-require */
-    const { buildConfig, changedFiles, sourcePaths, webhooks } = config;
+    const { buildConfig, changedFiles, sourcePaths, webhooks, prInfo } = config;
```
Comment (d2lam) [inline lib/eventFactory.js]: Need to change jsdoc to include `prInfo`

File: lib/eventFactory.js [hunk 4]
```diff
@@ -501,7 +510,8 @@ class EventFactory extends BaseFactory {
                             pipeline: p,
                             pipelineConfig: modelConfig,
                             changedFiles,
-                            webhooks
+                            webhooks,
+                            prInfo
```
Comment (d2lam) [inline lib/eventFactory.js]: Need to change jsdoc to include `prInfo` for this `create` function. 
This was probably missing from before

**Review by minzcmu — APPROVED** [review]

File: lib/build.js (hunk 1 above)
Comment (DekusDenial) [reply lib/build.js]: thx

Comment (DekusDenial) [reply lib/build.js]: thx

File: lib/eventFactory.js (hunk 2 above)
Comment (DekusDenial) [reply lib/eventFactory.js]: rename var to be more explicit

File: lib/eventFactory.js (hunk 4 above)
Comment (DekusDenial) [reply lib/eventFactory.js]: yup, just noticed that

**Pull request merged**